static/src/javascripts/projects/common/modules/identity/manage-account.js

```diff
@@ -37,17 +39,54 @@ const submitPartialFormStatus = (
     });
 };
 
+const getNewsletterHtmlPreferenceFromElement = (
+    originalEl: HTMLElement
+): Promise<string> =>
+    fastdom.read(() => {
+        const closestFormEl: ?Element = originalEl.closest('form');
+
+        if (!closestFormEl) throw Error(ERR_HTML_PREF_NOT_FOUND);
+
+        const checkboxEl: ?HTMLElement = closestFormEl.querySelector(
+            '[name="htmlPreference"]:checked'
+        );
+        const inputEl: ?HTMLElement = closestFormEl.querySelector(
+            '[name="htmlPreference"]'
+        );
+
+        if (checkboxEl && checkboxEl.value) {
+            return checkboxEl.value;
+        } else if (inputEl && inputEl.value) {
+            return inputEl.value;
+        }
+        throw Error(ERR_HTML_PREF_NOT_FOUND);
```

Comment:
`throw new Error()` ? I think I'd prefer the in another `else` branch, but that's only style I guess ...

Reply:
okay so i was actually super weirded out because i _remember_ putting this on an else branch  (I prefer them too) but turns out the linter eats them up